### ISSUE TITLE
what's the use of make_forecasting_frame?

### ISSUE THREAD

**Issue by vivi11130704** [blue-yonder/tsfresh#617]
1. Your operating system: 
win10
2. The version of *tsfresh* that you are using: 
0.13.0

### Question Summary ###
My understanding of creating df_shift from make_forecasting_frame is that the features can be extracted from the df_shift dataframe. However after looking at the extracted features, I realized the features are calculated directly from the raw dataframe without using df_shift at all. Can someone please explain the use of df_shift if it's not used for feature extraction?

### Question in detail ###

I'm working on a monthly revenue data for a commercial property. The goal is to use the past revenue to predict the future revenue. You can find my data [with this link](https://drive.google.com/file/d/1_4ejPljGAQouRLoNvfYtR5Glh1Ng12pr/view?usp=sharing). This data has 22 monthly data (from Jan 2018 - Oct 2019). Here is a quick screenshot of what it looks like:

![image](https://user-images.githubusercontent.com/7614915/71004789-6c358080-20b0-11ea-82fd-95decf4ee1bb.png)

I created a df_shift dataframe using the below code:
`df_shift, y = make_forecasting_frame(df['revenue'], kind="price", max_timeshift=6, rolling_direction=1)`

The output df_shft looks like this, which you can find [with this link](https://drive.google.com/open?id=1PJCK5v4jqJrVXY1KqWLekbXqYTCQhgfG)

![image](https://user-images.githubusercontent.com/7614915/71005072-fbdb2f00-20b0-11ea-83a0-cd6330549eeb.png)

Upon on this point, everything makes sense to me. Then I use this code to extract features:

`X = extract_features(df_shift, column_id="id", column_sort="time", column_value="value", n_jobs=0)`

The extracted features can be find [with this link](https://drive.google.com/open?id=1DCPFwXIiuAlkAsDDBLp0Q1RMIZZe-cEp). This feature df has 21 rows which matches the 22 months (minus the first month). Here is a quick screenshot:

![image](https://user-images.githubusercontent.com/7614915/71005643-ffbb8100-20b1-11ea-9eb1-729573835372.png)

**Question**: in feature screenshot id=3, the value_max is 269099. However looking at the df_shift screenshot id=3 (row 5-7), the value max is apparently 274875, which doesn't match. This means that the features are not extracted from df_shift corresponding windows. Rather, the feature is calculated from the adjacent two months from the original df. For instance, when id=3, value_max = 269099 is because it's comparing 3/1/2018 value with 2/1/2018 value (first screenshot, row 3-4). 

If features are not extracted from df_shift, why we spend so much effort in creating this df_shift? where it gets used? 

Thank you!

**Comment by nils-braun** [comment]
Hi @vivi11130704 

first of all thank you for this super nice bug-report. That is how all bug reports should be!

I am totally with you, the result looks really strange. Your code looks ok, though.

I can however assure you, that we do use the df_shift dataframe (because we do not have anything else) in the extract_features function, so this should work.

Actually I tried to redo what you did using the input data. I am using tsfresh version 0.13.0 as you do, but I am using Ubuntu 18.04 (but should hopefully! not make any different). And I get the result 274875.

So now things get confusing a bit :-)
Could you execute the attached notebook, which should reproduce it in a very minimal example just to be sure we are doing the same things here? I am quite sure your code is right, but I need to understand :-) 

 ```python3
csv_string = """
	id	time	revenue
0	113	2018-01-01	274875
1	113	2018-02-01	246107
2	113	2018-03-01	269099
3	113	2018-04-01	293351
4	113	2018-05-01	332037
5	113	2018-06-01	373034
6	113	2018-07-01	450922
7	113	2018-08-01	356766
8	113	2018-09-01	294907
9	113	2018-10-01	373487
10	113	2018-11-01	322669
11	113	2018-12-01	296076
12	113	2019-01-01	246868
13	113	2019-02-01	215942
14	113	2019-03-01	249878
15	113	2019-04-01	250822
16	113	2019-05-01	313437
17	113	2019-06-01	360334
18	113	2019-07-01	426831
19	113	2019-08-01	304587
20	113	2019-09-01	298255
21	113	2019-10-01	348617
"""

import pandas as pd
from io import StringIO

df = pd.read_csv(StringIO(csv_string), delim_whitespace=True)

from tsfresh.utilities.dataframe_functions import make_forecasting_frame

df_shift, y = make_forecasting_frame(df['revenue'], kind="price", max_timeshift=6, rolling_direction=1)

df_shift[df_shift.id == 3]

from tsfresh import extract_features
X = extract_features(df_shift, column_id="id",
                     column_sort="time", column_value="value",
                     n_jobs=0, default_fc_parameters={"maximum": None},
                     disable_progressbar=True)

print(X.head())

import tsfresh
print(tsfresh.__version__)

```

**Comment by nils-braun** [comment]
Just a small addon: I had a look into some more features of the extracted CSV you gave us and it seems they all have a length of 2. Could it by that you accidentally used `max_timeshift=2` before you did a feature extraction? Because that would exactly give your features :-)

**Comment by vivi11130704** [comment]
Hi @nils-braun 

Thanks so much for spending time to help me on this matter. Please see the below for the output: It seems like the output is correct now after directly writing out the dataframe, which is odd.. 

![image](https://user-images.githubusercontent.com/7614915/71010875-7bb9c700-20ba-11ea-9bb4-ad4f3585a172.png)

Here are some other things that I tested:
1. I also print out the df_shift that's produced by your code. it's exactly the same as the df_shift that's from my code. 
2. I add your code into my original code to replace the previous X = extract_features line that I have. Here is a screenshot of my code now:
![image](https://user-images.githubusercontent.com/7614915/71011645-c7b93b80-20bb-11ea-89af-70a7685f8381.png)

Here is the output, which is completely wrong and random. You can find the spreadsheet [with this link](https://drive.google.com/open?id=1TOSlK7j61vSUm4o7y_sZf0aiWbplF2va)

![image](https://user-images.githubusercontent.com/7614915/71011696-de5f9280-20bb-11ea-91b9-c1d13f1703b5.png)

**Comment by vivi11130704** [comment]
Hi @nils-braun 

Thanks so much for your time. I have figured out the problem - I used your code and the extracted features was correct. So I tried to feed in a new set of data to my original program for testing again. Then I realize my pycharm were not able to read the new data, it keeps returning the old dataframe. So I closed my pycharm and reopen it. Now everything works as expected, the extracted features are correct now. I don't know why this happens. This is very odd. But I'm glad the tsfresh is working great now.

Thank you so much and sorry for the confusion!
